### ISSUE TITLE
Cop idea: `RedundantSuperArguments`

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#12309]
## Is your feature request related to a problem? Please describe.

Consider the following code:

```rb
def input(attribute_name, options = {}, &)
  value = @options[attribute_name]
  options[:value] = value
  super(attribute_name, options, &)
end
```

Ruby delegates all arguments to super if called without anything. Rubocop should detect this case and autocorrect to a plain super call.

## Describe the solution you'd like
Autocorrect to the following if the arguments to super matches the method signature:

```rb
def input(attribute_name, options = {}, &)
  value = @options[attribute_name]
  options[:value] = value
  super
end
```

## Additional context

Calling super with parents like `super()` has special meaning and will not pass any arguments to the inherited method. There should be no autocorrect in that case.
